src/core/lv2/Lv2Proc.cpp

```diff
@@ -168,13 +168,35 @@ Plugin::Type Lv2Proc::check(const LilvPlugin *plugin,
 
 
 
+class Lv2ProcSuspender
+{
+public:
+	Lv2ProcSuspender(Lv2Proc* proc)
+		: m_proc(proc)
+		, m_wasActive(proc->m_instance)
+	{
+		if (m_wasActive) { m_proc->shutdownPlugin(); }
+	}
+	~Lv2ProcSuspender()
+	{
+		if (m_wasActive) { m_proc->initPlugin(); }
+	}
+private:
+	Lv2Proc* const m_proc;
+	const bool m_wasActive;
```

Comment:
Must these variables be `const`?

Reply:
I understand the Guidelines as [`const` should always be preferred](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable)?

Reply:
If the intent is to initialize the class with the value and it will never change within the class for any reason, then `const` is fine.

Reply:
I generally do not make member variables `const` because it breaks the general semantics of the class, such as moving it.

Reply:
I still prefer having `const`-safety over not break copyability which I actually even want to break.

Actually, I even added this on purpose. I copied my code from

https://github.com/LMMS/lmms/blob/6d4343ca948d81d30698cf9442c84fc595bd2bed/plugins/VstBase/RemoteVstPlugin.cpp#L401-L417

and I was first confused about this code. I was somehow thinking `m_resumed` was something like a state which changes when the plugin gets resumed - `const` makes this clear.

Reply:
> I still prefer having const-safety over not break copyability which I actually even want to break.

If you do not intend this class to be copied, movied, or assigned, I would disable the operations explicitly with `= delete`, because now its not as clear that this class is not meant to be copied, moved or assigned in any way. I understand this class to be a scope guard, and scope guards (atleast in the TS) do not have such capabilities for the most part.